src/Nethermind/Nethermind.TxPool/INonceManager.cs

```diff
@@ -0,0 +1,15 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.TxPool;
+
+public interface INonceManager
+{
+    IDisposable ReserveNonce(Address address, out UInt256 nonce);
+    IDisposable TxWithNonceReceived(Address address, UInt256 nonce);
+    void TxAccepted(Address address);
```

Comment:
API: I would make the result of `ReserveNonce` and `TxWithNonceReceived` contextual, I would move `Accept` method on that object (make it a `struct` or even better `ref struct` too)